index.bs

```diff
@@ -371,28 +403,159 @@ model]]:
+
+Insert the following steps in the <a spec="html">navigate</a> algorithm before step 8,
+"[Navigate to a fragment](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate-fragid) given navigable, url, historyHandling, and navigationId."
+
+1. Run [=process navigation start for bounce tracking=] given
+    <var ignore>navigable</var>, <var ignore>sourceDocument</var>, and <var ignore>sourceSnapshotParams</var>.
+
+<div algorithm>
+
+To <dfn>process navigation start for bounce tracking</dfn> given a [=navigable=]
+|navigable|, [=Document=] |sourceDocument|, and <a spec="html">source snapshot params</a> |sourceSnapshotParams|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. Let |origin| be |sourceDocument|'s [=Document/origin=].
+1. If |origin| is an [=opaque origin=], set |initialHost| to empty string.
+1. Otherwise,
+    1. Let |site| be the result of running [=obtain a site=] given |origin|.
+    1. Set |initialHost| to |site|'s [=host=].
+1. If |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] is false:
+    1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new
+        [=bounce tracking record=] with [=bounce tracking record/initial host=]
+        set to |initialHost|.
+1. Otherwise,
+    1. If |sourceSnapshotParams|'s <a spec="html">has transient activation</a> is true:
+        1. Run [=record stateful bounces for bounce tracking=].
+        1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new [=bounce tracking record=] with
+            [=bounce tracking record/initial host=] set to |initialHost|.
+    1. Otherwise, add |initialHost| to |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=].
+1. Set |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] to false.
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-response-received-monkey-patch">Response Received Monkey
+Patch</h5>
+
+When the response is received at the end of a navigation, fill the [=bounce tracking record/bounce set=].
+
+Insert the following steps in the <a spec="html">navigate</a> algorithm, under "Create navigation params by fetching", after Step 19.7,
+"Wait until either response is non-null...", or as soon as [=response/url list=] becomes available.
+
+1. Run [=process response received for bounce tracking=] given
+    <var ignore>navigable</var> and <var ignore>response</var>'s <var ignore>URL list</var>.
+
+<div algorithm>
+
+To <dfn>process response received for bounce tracking</dfn> given a [=navigable=]
+|navigable| and a [=response/url list=] |URLs|, perform the following steps:
+
+1. [=Assert=] that |navigable|'s [=top-level traversable/bounce tracking record=]'s
+    [=bounce tracking record/client bounce detection running=] is false.
```

Comment:
This being false will cause "record stateful bounces for bounce tracking" to always bail out early.

Reply:
Not sure why this is here, I think this was meant for the "record stateful bounces for bounce tracking" algorithm. Removed.